File: pubnub-gson/pubnub-gson-api/src/main/java/com/pubnub/api/java/endpoints/datasync/entity/UpdateEntity.java
package com.pubnub.api.java.endpoints.datasync.entity;

import com.pubnub.api.java.endpoints.Endpoint;
import com.pubnub.api.java.models.consumer.datasync.entity.PNDataSyncUpdateEntityResult;
import org.jetbrains.annotations.Nullable;

/**
 * @see com.pubnub.api.java.datasync.DataSync#updateEntity(String, List)
 */
public interface UpdateEntity extends Endpoint<PNDataSyncUpdateEntityResult> {
    /**
     * Optional eTag for optimistic concurrency ({@code If-Match} header).
     */
    UpdateEntity ifMatch(@Nullable String ifMatch);
}

File: pubnub-kotlin/pubnub-kotlin-api/src/jvmMain/kotlin/com/pubnub/api/endpoints/datasync/entity/UpdateEntity.kt
package com.pubnub.api.endpoints.datasync.entity

import com.pubnub.api.Endpoint
import com.pubnub.api.models.consumer.datasync.entity.DataSyncUpdateEntityResult

/**
 * @see [com.pubnub.api.datasync.DataSync.updateEntity]
 */
actual interface UpdateEntity : Endpoint<DataSyncUpdateEntityResult>
